Assets/Scripts/Scriptable Objects/ChargeBehaviour.cs

```diff
@@ -0,0 +1,86 @@
+    private float timeSinceLastCharge = 0;
+    private bool charging = false;
+    private Collider hitbox;
+    private bool hitPlayer = false;
+
+    public override void Act(ref Knowledge knowledge)
+	{
+        if (charging)
+        {
+            knowledge.Agent.isStopped = false;
+            // If charging, keep trying to collide with the player until it happens or the charge is cancelled.
+            if (!hitPlayer)
+            {
+                // Collides with player. Feels a bit dodgy since it might not perfectly reflect the furniture's total hitbox but it seemed more responsive at the time than using the furniture's own collider trigger.
+                // Anyone can feel free to change this if there are any problems.
+                Collider[] hitColliders = Physics.OverlapBox(hitbox.bounds.center, hitbox.bounds.size, knowledge.AITransform.rotation);
+
+                foreach (Collider hitCollider in hitColliders)
+                {
+                    // If the player manages to collide with the furniture, deal damage and apply knockback! Make sure this only happens once per charge.
+                    if (hitCollider.CompareTag("Player"))
+                    {
+                        HuntingManager.Instance.DealDamageToPlayer(damage);
+                        hitPlayer = true;
+                        hitCollider.transform.parent.GetComponent<Player>().ApplyKnockback(new Vector3(knowledge.Agent.velocity.x * horizontalKnockbackMultiplier, verticalKnockback, knowledge.Agent.velocity.z * horizontalKnockbackMultiplier));
+                        break;
+                    }
+                }
+            }
+            // If the agent finishes pathfinding, or the player's position is out of range AND somewhere behind the furniture, stop the charge.
+            Vector3 targetDir = knowledge.PlayerPosition - knowledge.AITransform.position;
+            float angle = Vector3.Angle(targetDir, knowledge.AITransform.forward);
+            if (knowledge.Agent.remainingDistance < 1f || (Vector3.Distance(knowledge.AITransform.position,knowledge.PlayerPosition) > turnRange && angle > 120))
+            {
+                charging = false;
+                knowledge.Agent.SetDestination(knowledge.AITransform.position); // Somehow relying on isStopped doesn't work. This code however, DOES work!
+            }
+        }
+        else
+        {
+            // Make sure the furniture is stationary and always turning to face the player, counting on a timer.
+            knowledge.Agent.isStopped = true;
+            Quaternion lookRotation = Quaternion.LookRotation((new Vector3(knowledge.PlayerPosition.x, knowledge.AITransform.position.y, knowledge.PlayerPosition.z) - knowledge.AITransform.position).normalized);
+            knowledge.AITransform.rotation = Quaternion.Slerp(knowledge.AITransform.rotation, lookRotation, Time.deltaTime * 5);
```

Comment:
Avoid this harcoded 5, expose it so different furniture can have different turning rates